crates/cargo-contract/src/cmd/metadata.rs

```diff
@@ -73,6 +75,42 @@ struct ExtendedMetadataResult {
     user: Option<User>,
 }
 
+/// Information about the settings used to build a particular ink! contract.
+///
+/// Note that this should be an optional part of the metadata since it may not necessarily
+/// translate to other languages (e.g ask!, Solidity).
+#[derive(Clone, Debug, serde::Deserialize, serde::Serialize)]
+pub struct BuildInfo {
+    /// The stable version of `rustc` used to build the contract.
+    pub rustc_version: Version,
```

Comment:
Can/should we also add the toolchain which contains the architecture? e.g. `stable-x86_64-unknown-linux-gnu`. I remember there being some discussions about this affecting the wasm bytecode output, can't remember what the conclusions were.